app/src/main/java/fr/free/nrw/commons/nearby/NearbyController.java

```diff
@@ -255,4 +226,18 @@ public NearbyPlacesInfo loadAttractionsFromLocation(LatLng curLatLng, LatLng sea
         public LatLng curLatLng; // Current location when this places are populated
         public LatLng searchLatLng; // Search location for finding this places
     }
+
+    /**
+     * Updates makerLabelList item isBookmarked value
+     * @param place place which is bookmarked
+     * @param isBookmarked true is bookmarked, false if bookmark removed
+     */
+    public static void updateMarkerLabelListBookmark(Place place, boolean isBookmarked) {
+        for (ListIterator<MarkerPlaceGroup> iter = markerLabelList.listIterator(); iter.hasNext();) {
```

Comment:
agh static lists! I would be very concerned about ConcurrentModification, this is not thread safe for ArrayList.

Maybe a CopyOnWriteArrayList would be best here? I am really not sure without diving deeper

Reply:
Hmmm actually I think copyOnWrite iterators don't allow modification... maybe we can mark the methods that update the list `@MainThread` which I am assuming is the assumption that keeps this working?